server/src/controllers/organizations.controller.ts

```diff
@@ -30,11 +32,42 @@ export class OrganizationsController {
     private readonly configService: ConfigService
   ) {}
 
-  @UseGuards(JwtAuthGuard)
+  @UseGuards(JwtAuthGuard, IsAdminGuard)
   @Get('users')
-  async getUsers(@Request() req) {
-    const result = await this.organizationsService.fetchUsers(req.user);
-    return decamelizeKeys({ users: result });
+  async getUsers(@User() user, @Query() query) {
+    const { page, email, firstName, lastName } = query;
+    const filterOptions = {
+      ...(email && { email }),
+      ...(firstName && { firstName }),
+      ...(lastName && { lastName }),
+    };
+    const users = await this.organizationsService.fetchUsers(user, page, filterOptions);
+    const usersCount = await this.organizationsService.usersCount(user, filterOptions);
```

Comment:
Shall we first take the user count and if its greater than zero go for fetch users